src/udev/udev-rules.c

```diff
@@ -1338,9 +1338,8 @@ bool udev_rules_check_timestamp(UdevRules *rules) {
                 if (stat(*f, &st) < 0)
                         continue;
 
-                r = hashmap_put_stats_by_path(&stats_by_path, *f, &st);
-                if (r < 0)
-                        return r;
+                if (hashmap_put_stats_by_path(&stats_by_path, *f, &st) < 0)
+                        return true;
```

Comment:
maybe add a comment that errors are eaten up here and why that's safe. and maybe add log_debug() about unexpected erros.